### PR TITLE
[DOCS] Adds 7.3.0 ml-cpp release notes

### PR DESCRIPTION
**Pull request by lcawl**
This PR adds the PRs fromhttps://raw.githubusercontent.com/elastic/ml-cpp/7.3/docs/CHANGELOG.asciidoc to the Elasticsearch Reference 7.3.0 Release Notes.

It also edits the existing machine learning and data frame transform PRs.

### CODE REVIEW

File: docs/reference/release-notes/7.3.asciidoc [hunk 1]
```diff
@@ -276,19 +276,19 @@ Cluster Coordination::
...
+* Adds index validations to _start data frame transform {pull}44191[#44191] (issue: {issue}44104[#44104])
+* Data frame task failure do not make a 500 response {pull}44058[#44058] (issue: {issue}44011[#44011])
+* Audit message missing for autostop {pull}43984[#43984] (issue: {issue}43977[#43977])
+* Add data frame transform cluster privileges to HLRC {pull}43879[#43879]
+* Fix progress measurement for continuous transforms {pull}43838[#43838]
+* Improve pivot nested field validations {pull}43548[#43548]
+* Adjusting error message {pull}43455[#43455]
+* Size the GET stats search by number of Ids requested {pull}43206[#43206] (issue: {issue}43203[#43203])
+* Rewrite start and stop to answer with acknowledged {pull}42589[#42589] (issue: {issue}42450[#42450])
+* Set DF task state to stopped when stopping   {pull}42516[#42516] (issue: {issue}42441[#42441])
```
Comment (droberts195) [inline docs/reference/release-notes/7.3.asciidoc]: Is the "DF" abbreviation acceptable to use?

File: docs/reference/release-notes/7.3.asciidoc [hunk 2]
```diff
@@ -276,19 +276,19 @@ Cluster Coordination::
...
+* Add support for fixed_interval, calendar_interval, remove interval {pull}42427[#42427] (issues: {issue}33727[#33727], {issue}42297[#42297])
```
Comment (droberts195) [inline docs/reference/release-notes/7.3.asciidoc]: This went into 7.2.0, which was when data frames was first released.  From an end user's perspective data frame transforms have always been like this, so this item can be removed from the release notes.

File: docs/reference/release-notes/7.3.asciidoc (hunk 1 above)
Comment (hendrikmuhs) [reply docs/reference/release-notes/7.3.asciidoc]: +1, "DF" seems superflous.

File: docs/reference/release-notes/7.3.asciidoc [hunk 3]
```diff
@@ -276,19 +276,19 @@ Cluster Coordination::
...
-* [ML-DataFrame] rewrite start and stop to answer with acknowledged {pull}42589[#42589] (issue: {issue}42450[#42450])
-* [ML Data Frame] Set DF task state to stopped when stopping   {pull}42516[#42516] (issue: {issue}42441[#42441])
-* [ML-DataFrame] add support for fixed_interval, calendar_interval, remove interval {pull}42427[#42427] (issues: {issue}33727[#33727], {issue}42297[#42297])
+* Treat bulk index failures as an indexing failure {pull}44351[#44351] (issue: {issue}44101[#44101])
+* Responding with 409 status code when failing _stop {pull}44231[#44231] (issue: {issue}44103[#44103])
+* Adds index validations to _start data frame transform {pull}44191[#44191] (issue: {issue}44104[#44104])
+* Data frame task failure do not make a 500 response {pull}44058[#44058] (issue: {issue}44011[#44011])
+* Audit message missing for autostop {pull}43984[#43984] (issue: {issue}43977[#43977])
+* Add data frame transform cluster privileges to HLRC {pull}43879[#43879]
+* Fix progress measurement for continuous transforms {pull}43838[#43838]
```
Comment (hendrikmuhs) [inline docs/reference/release-notes/7.3.asciidoc]: we should remove this, progress is broken for continuous

File: docs/reference/release-notes/7.3.asciidoc [hunk 4]
```diff
@@ -180,20 +178,22 @@ Infra/Scripting::
...
-* [ML] Better detection of binary input in find_file_structure {pull}42707[#42707]
-* [ML] Add a limit on line merging in find_file_structure {pull}42501[#42501] (issue: {issue}38086[#38086])
+* Better detection of binary input in find_file_structure {pull}42707[#42707]
+* Add a limit on line merging in find_file_structure {pull}42501[#42501] (issue: {issue}38086[#38086])
 * Implement factory methods for ValidationException {pull}41993[#41993]
-* [ML] Improve file structure finder timestamp format determination {pull}41948[#41948] (issues: {issue}35132[#35132], {issue}35137[#35137], {issue}38086[#38086])
+* Improve file structure finder timestamp format determination {pull}41948[#41948] (issues: {issue}35132[#35132], {issue}35137[#35137], {issue}38086[#38086])
 * Increase maximum forecast interval to 10 years. {pull}41082[#41082] (issue: {issue}41103[#41103])
+* Upgrade to a newer version of the Apache Portable Runtime library. {ml-pull}495[#495]
+* Improve stability of modelling around change points. {ml-pull}496[#496]
```
[the remark this answers is not in the document]
Comment (lcawl) [reply docs/reference/release-notes/7.3.asciidoc]: It's not covered in the Style Guide, so I think it's fine to leave it as-is

**Comment by lcawl** [comment]
Thanks for all the feedback, @droberts195 and @hendrikmuhs

**Review by droberts195 — APPROVED** [review]
LGTM

**Pull request merged**